Processor/ProductToFlatArrayProcessor.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace Pim\Bundle\EnhancedConnectorBundle\Processor;
+
+
+use Akeneo\Bundle\BatchBundle\Item\ItemProcessorInterface;
+use Pim\Bundle\BaseConnectorBundle\Processor\ProductToFlatArrayProcessor as ProductToFlatArray;
+use Pim\Bundle\CatalogBundle\Doctrine\ORM\Repository\AssociationTypeRepository;
+use Pim\Bundle\CatalogBundle\Doctrine\ORM\Repository\AttributeGroupRepository;
+use Pim\Bundle\CatalogBundle\Entity\AssociationType;
+use Pim\Bundle\CatalogBundle\Entity\AttributeGroup;
+use Pim\Bundle\CatalogBundle\Manager\ChannelManager;
+use Pim\Bundle\CatalogBundle\Model\ProductValueInterface;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * Class ProductToFlatArrayProcessor
+ *
+ * @author  Synolia
+ * @package Pim\Bundle\EnhancedConnectorBundle\Processor
+ */
+class ProductToFlatArrayProcessor extends ProductToFlatArray implements ItemProcessorInterface
+{
+    /** @var array */
+    protected $attributesToExclude;
+
+    /** @var  array */
+    protected $associationsToExclude;
+
+    /** @var  AttributeGroupRepository */
+    protected $attributeGroupRepository;
+
+    /** @var  AssociationTypeRepository */
+    protected $associationTypeRepository;
+
+    /**
+     * @param Serializer $serializer
+     * @param ChannelManager $channelManager
+     * @param AttributeGroupRepository $attributeGroupRepository
+     * @param AssociationTypeRepository $associationTypeRepository
+     * @param string[] $mediaAttributeTypes
+     */
+    public function __construct(
+        Serializer $serializer,
+        ChannelManager $channelManager,
+        AttributeGroupRepository $attributeGroupRepository,
+        AssociationTypeRepository $associationTypeRepository,
+        array $mediaAttributeTypes
+    ) {
+        parent::__construct($serializer, $channelManager, $mediaAttributeTypes);
+
+        $this->attributeGroupRepository  = $attributeGroupRepository;
+        $this->associationTypeRepository = $associationTypeRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process($product)
```

Comment:
I think the best strategy would be to simply call `parent::process()` and then filter out the produced array to remove the unwanted field.

This way, you keep the normal behavior of the processor, and benefit from all the new stuff that are coming with new versions.